src/driver.ml

```diff
@@ -95,6 +95,14 @@ module Instrument = struct
 end
 
 module Transform = struct
+  type kind =
+    | WholeFileTransform
+    | BuiltinContextFree
+    | BuiltinLinter
+    | BuiltinPreprocess
+    | BuiltinBeforeInstr
+    | BuiltinAfterInstr
```

Comment:
I would say that whole-file transformations are all transformations that transform the whole AST (as opposed to context-free rules, that just extend one node), so that includes instrumentation, linters and preprocess. Since you have some kinds of whole-file-transformations separately, it might be a bit confusing to have one variant called `WholeFileTransform`. Is it possible to separate it into `Impl` (whole-implementation-file-transformations) and `Intf` (whole-interface-file-transformations)?